input/config.js

```diff
@@ -1,103 +1,161 @@
+/**************************************************************
+ * UTILITY FUNCTIONS
+ * - scroll to BEGIN CONFIG to provide the config values
+ *************************************************************/
 const fs = require("fs");
-const width = 1000;
-const height = 1000;
 const dir = __dirname;
-const description = "This is an NFT made by the coolest generative code.";
-const baseImageUri = "https://hashlips/nft";
-const startEditionFrom = 1;
-const editionSize = 10;
-const rarityWeights = [
-  {
-    value: "super_rare",
-    from: 1,
-    to: 1,
-  },
-  {
-    value: "rare",
-    from: 2,
-    to: 5,
-  },
-  {
-    value: "original",
-    from: 5,
-    to: editionSize,
-  },
-];
 
+// adds a rarity to the configuration. This is expected to correspond with a directory containing the rarity for each defined layer
+// @param _id - id of the rarity
+// @param _from - number in the edition to start this rarity from
+// @param _to - number in the edition to generate this rarity to
+// @return a rarity object used to dynamically generate the NFTs
+const addRarity = (_id, _from, _to) => {
+  const _rarityWeight = {
+    value: _id,
+    from: _from,
+    to: _to,
+    layerPercent: {}
+  };
+  return _rarityWeight;
+};
+
+// get the name without last 4 characters -> slice .png from the name
 const cleanName = (_str) => {
   let name = _str.slice(0, -4);
   return name;
 };
 
-const getElements = (path) => {
+// reads the filenames of a given folder and returns it with its name and path
+const getElements = (_path, _elementCount) => {
   return fs
-    .readdirSync(path)
+    .readdirSync(_path)
     .filter((item) => !/(^|\/)\.[^\/\.]/g.test(item))
     .map((i) => {
       return {
+        id: _elementCount,
         name: cleanName(i),
-        path: `${path}/${i}`,
+        path: `${_path}/${i}`
       };
     });
 };
 
+// adds a layer to the configuration. The layer will hold information on all the defined parts and 
+// where they should be rendered in the image
+// @param _id - id of the layer
+// @param _position - on which x/y value to render this part
+// @param _size - of the image
+// @return a layer object used to dynamically generate the NFTs
+const addLayer = (_id, _position, _size) => {
+  if (!_id) {
+    console.log('error adding layer, parameters id required');
+    return null;
+  }
+  if (!_position) {
+    _position = { x: 0, y: 0 };
+  }
+  if (!_size) {
+    _size = { width: width, height: height }
+  }
+  // add two different dimension for elements:
+  // - all elements with their path information
+  // - only the ids mapped to their rarity
+  let elements = [];
+  let elementCount = 0;
+  let elementIdsForRarity = {};
+  rarityWeights.forEach((rarityWeight) => {
+    let elementsForRarity = getElements(`${dir}/${_id}/${rarityWeight.value}`);
+
+    elementIdsForRarity[rarityWeight.value] = [];
+    elementsForRarity.forEach((_elementForRarity) => {
+      _elementForRarity.id = `${editionDnaPrefix}${elementCount}`;
```

Comment:
The dna now uses an id that spans all elements within a layer and is not specific to a rarity anymore. This allows to additionally differentiate the dna (since there would be duplication otherwise), but primarily is used to allow for using elements from any rarity during the creation process. 
Also added an `editionDnaPrefix` that can be used to further differentiate the dna, i.e. if a user would create a second batch of NFTs for the same edition using different input files.